lib/search/client/elasticsearch.rb

```diff
@@ -1,77 +1,85 @@
 require "search/client"
+require 'search/client/search_result'
+require 'elasticsearch'
 
 class Search
   class Client
     class Elasticsearch < Search::Client
+      attr_accessor :client, :index_name, :settings, :type
       ESCAPE_LUCENE_CHARS = /
         ( [-+!\(\)\{\}\[\]^"~*?:\\] # A special character
           | &&                        # Boolean &&
           | \|\|                      # Boolean ||
         )/x
 
-      def initialize(config)
-        super
+      def initialize(index_name:, settings:, type:, config:)
+        super(config)
+        @index_name = index_name
+        @settings = settings
+        @type = type
         @config = config
-        configure
-      end
-
-      def configure
-        url = @config[:url]
-        Tire.configure { url url }
-        logger = @config[:logger]
-        if logger
-          Tire.configure { logger logger }
-        end
-      end
-
-      def indexer
-        return @indexer unless @indexer.nil?
-        @indexer = Tire::Index.new(@config[:index])
+        @client = ::Elasticsearch::Client.new(config)
       end
 
       def delete_index
-        indexer.delete
+        if client.indices.exists?(index: index_name)
```

Comment:
(Alternatively, I think you can possibly do something like `client.indices.delete(index: index_name, ignore: [404]` This is possibly a bit more obscure, but halves the number of requests you're making. See: https://github.com/elastic/elasticsearch-ruby/blob/master/elasticsearch-api/lib/elasticsearch/api/actions/indices/delete.rb#L41-L42 )